src/components/Firestore/Document.tsx

```diff
@@ -26,41 +26,59 @@ import DocumentPreview from './DocumentPreview';
 import Collection from './Collection';
 import { useApi } from './ApiContext';
 
-export interface Props {
-  reference?: firestore.DocumentReference;
-  snapshot?: firestore.DocumentSnapshot;
-}
-
-export const Document: React.FC<Props> = ({ reference, snapshot }) => {
+const Doc: React.FC<{
+  id: string;
+  selectionGetter: (id: string) => firestore.CollectionReference;
```

Comment:
nit: rename to `collectionById`? The current name is confusing to me.

Reply:
Done